cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/ExampleFlow.kt

```diff
@@ -104,9 +111,8 @@ object ExampleFlow {
             val signTransactionFlow = object : SignTransactionFlow(otherPartySession) {
                 override fun checkTransaction(stx: SignedTransaction) = requireThat {
                     // The transaction involves an IOUState - this ensures that IOUContract will be run to verify the transaction
-                    val output = stx.tx.outputs.single().data
-                    "This must be an IOU transaction." using (output is IOUState)
-                    val iou = output as IOUState
+                    "This must be an IOU transaction." using (stx.tx.outputsOfType<IOUState>().size == 1)
+                    val iou = stx.tx.outputsOfType<IOUState>().single()
```

Comment:
Still not checking if my signature is meaningful.

Reply:
You should also check that the amount in the Cash state matches the amount in the IOU state, otherwise the lender can say hey I lended you 1000,000$ while they will only send you 1$!

Reply:
Indeed, the flow is the right place to do this check. After all, if Alice repays to Bob a 100$ loan with 1$, the ledger is not broken. On the other hand, we understand that Bob is unlikely to accept, so that's in the flow.
To further the point, I will say that the IOU state / contract need to be able to account for Alice's bankruptcy. In this case, a new flow could be created where Bob will accept 10$ instead of 100$. Or even 1$ or 0$. If this was not possible, then this IOUState instance would live forever without reason.
If you absolutely wanted the IOU contract to control the amounts, that adds a lot of complexity. And the contract would have to introduce a "bankruptcy" exit too.

Reply:
I DO appreciate your comments, I really am, because it shows me how I underestimated this exercise. 
But I feel like I never move forward if I will be adding new features to this very first exercise :) I promise I will do much better in next module. 

about checking signature for `ExampleFlow`, in IOU contract there are two constrains:
```
"Command require two signers." using (command.signers.size == 2)
                "All of the participants must be signers." using (command.signers.containsAll(listOf(out.lender.owningKey, out.borrower.owningKey)))
```
aren't they enough for checking signature?

Reply:
Thank you.

Checking for these 2 signatures is good enough for the "signatures checking" part.